python/functional-programming/sum-all-map-functions/the-map-built-in.md

```diff
@@ -29,92 +18,35 @@ revisionQuestion:
-
-*Remember, in functional programming, functions can take functions. So, instead of a list of data, `map` may take a list of functions, and output a list of new functions. This gives a sense of how versatile and useful the `map` function can be.*
-
-
----
-
-## Practice
-
-Let's say we have a list, called `promises`. We want to `make_good` on all our promises, where `make_good` is a previously-defined function that takes a string. Fill in the blanks in the code below to apply `make_good` to all elements in `promises`.  
-
-```python
-promises = ['learn css', 'learn js', 'buy milk', 'be excellent to each other']
-promises = ???(???, ???)
-```
-
-- `map`
-- `make_good`
-- `promises`
-- `lambda x: x*2`
-- list
-- reduce
-- filter
-
-
----
+The `map` built-in applies a given function to all elements of a given iterable. 
 
-## Revision
+This allows us to avoid imperative programming[1], often resulting in more easily legible code. 
 
-We are using the map function on two lists of integers at the same time. What is the output from the following code snippet?
+In the functional programming style[2], we can achieve the previous code with:
 
 ```python
+def addTwo(x):
+  return x + 2
 scores = [1, 0, 4, 5]
-scores2 = [1, 1, 1, 1]
-newscores = map(
-  lambda x, y : x+y, scores, scores2)
-print(newscores)
+print(list(map(addTwo, scores)))
+# Result: [3,2,6,7]
 ```
 
-???
 
-- `[2, 1, 5, 6]`
-- `[6, 5, 1, 2]`
-- `[1, 0, 4, 5, 1, 1, 1, 1]`
-- `[1, 1, 1, 1, 1, 0, 4, 5]`
+> 🥳 Today is 5 of your users' birthdays! Will you use a `for` loop or the `map` function to increment their ages?
 
 
 ---
 
 ## Footnotes
 
-[1:input_list]
-As long as our `input_list` argument given to `map` is of type `iterable`, `map` should work. To check this you can use duck typing or type checking of Abstract Base Classes (ABCs)
 
-[2:Imperative Programming]
-Imperative programming, as opposed to declarative programming (of which functional programming is a subtype) is the programming paradigm where statements are used to change a program's state. Imperative style requires certain mechanisms such as direct data modification (mutability) and looping statements. These mechanisms sometimes detrimentally effect readability and efficiency.
+[1:Imperative Programming]
+Imperative programming, as opposed to declarative programming (of which functional programming is a subtype) is the programming paradigm where statements are used to change a program's state. Imperative style requires certain mechanisms such as direct data modification (mutability) and looping statements. These mechanisms sometimes detrimentally affect readability and efficiency.
 
 An easy rule of thumb is that imperative programming specifies **how** (what steps are taken during execution) a program should operate. This is in contrast to declarative languages which focus on **what** is to be computed instead.
```

Comment:
We might not need to repeat this footnote since we already touch upon it in an earlier insight. Or we can copy paste text from the other footnote into this one.

We do have a way of adding global footnotes via the glossary as well (reach out to @Stefan-Stojanovic for details)